lib/drivers/mock/mock.go

```diff
@@ -6,21 +6,32 @@ import (
 
 type MockDriver struct {
 	Revisions []core.Revision
-	State     core.Revision
 }
 
 func (this *MockDriver) Init(config map[string]string) error {
 	return nil
 }
 
-func Migrate(r core.Revision, ec chan error) {
+func (this *MockDriver) Migrate(r core.Revision, ec chan error) {
+	for _, mig := range r.Migrate {
+		if _, err = mig; err != nil {
+			ec <- err
+			return
+		}
+	}
 }
 
-func Rollback(r core.Revision, ec chan error) {
+func (this *MockDriver) Rollback(r core.Revision, ec chan error) {
+	for _, mig := range r.Rollback {
+		if _, err = mig; err != nil {
+			ec <- err
+			return
+		}
+	}
 }
 
-func State() *core.Revision {
-	return &core.Revision{}
+func (this *MockDriver) State() (*core.Revision, error) {
+	return nil, &core.Revision{}
```

Comment:
This is the wrong ordering for the return value. Since you pulled "Revision from the MockDriver you will need to return the last element of they array.